src/vector.js

```diff
@@ -53,6 +154,17 @@ Vector.cross  = function (v1, v2)
     ];
 };
 
+// Dot product of two vectors
+Vector.dot = function (v1, v2)
+{
+    var n = 0.0;
+    var lim = Math.min(v1.length, v2.length);
```

Comment:
I agree with Brett, it seems fine to just use the length of the first argument. One thing to mention is that JavaScript Arrays do not throw out of bound errors for element they don't have. They return undefined, so 

```javascript
var a = [1,2,3,4];
a[10] === undefined; // true
```
So if you do have a shorter Array as the first argument, the multiplication operation will produce a 'NaN'.